### ISSUE TITLE
AggregateWindowFunction needs to call BaseVector::prepareForReuse

### ISSUE THREAD

**Issue by mbasmanova** [facebookincubator/velox#3304]
Aggregate function expects all-flat 'result' vector, but may produce a non-flat result. For example, map_agg and map_union return dictionary encoded keys if there are duplicates.

AggregateWindowFunction needs to call BaseVector::prepareForReuse(aggregateResultVector_) before calling aggregate_->extractValues.

See example in HashAggregation:

```
void HashAggregation::prepareOutput(vector_size_t size) {
  if (output_) {
    VectorPtr output = std::move(output_);
    BaseVector::prepareForReuse(output, size);
    output_ = std::static_pointer_cast<RowVector>(output);
  } else {
    output_ = std::static_pointer_cast<RowVector>(
        BaseVector::create(outputType_, size, pool()));
  }
}
```

See #3291 for repro.

**Comment by aditi-pandit** [comment]
Assigning to @mbasmanova to close.

**Comment by mbasmanova** [comment]
Fixed in #3291
